src/server/command/worldedit/blueprint.zig

```diff
@@ -0,0 +1,193 @@
+const std = @import("std");
+
+const main = @import("root");
+const User = main.server.User;
+const vec = main.vec;
+const Vec3i = vec.Vec3i;
+
+const List = main.List;
+const Block = main.blocks.Block;
+const Blueprint = main.blueprint.Blueprint;
+const NeverFailingAllocator = main.heap.NeverFailingAllocator;
+
+pub const description = "Operate on blueprints.";
+pub const usage = "/blueprint <save FILENAME|delete FILENAME|load FILENAME|list>";
+
+const BlueprintSubCommand = enum {
+	save,
+	delete,
+	load,
+	list,
+	other,
+	empty,
+
+	fn fromString(string: []const u8) BlueprintSubCommand {
+		return std.meta.stringToEnum(BlueprintSubCommand, string) orelse {
+			if(string.len == 0) return .empty;
+			return .other;
+		};
+	}
+};
+
+pub fn execute(args: []const u8, source: *User) void {
+	var argsList = List([]const u8).init(main.stackAllocator);
+
+	var splitIterator = std.mem.splitSequence(u8, args, " ");
+	while(splitIterator.next()) |a| {
+		argsList.append(main.stackAllocator.dupe(u8, a));
```

Comment:
duping is not needed, since `args` (which `a` points into) lives for the entire duration of the function.